src/deck.js

```diff
@@ -0,0 +1,82 @@
+ * @copyright Copyright (c) 2020 Vincent Petry <vincent@nextcloud.com>
+ *
+ * @author Vincent Petry <vincent@nextcloud.com>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import Vue from 'vue'
+import { generateFilePath, generateOcsUrl, generateUrl } from '@nextcloud/router'
+import { getRequestToken } from '@nextcloud/auth'
+import { translate, translatePlural } from '@nextcloud/l10n'
+import { showSuccess, showError } from '@nextcloud/dialogs'
+import SHA1 from 'crypto-js/sha1'
+import Hex from 'crypto-js/enc-hex'
+import axios from '@nextcloud/axios'
+
+// CSP config for webpack dynamic chunk loading
+// eslint-disable-next-line
+__webpack_nonce__ = btoa(getRequestToken())
+
+// Correct the root of the app for chunk loading
+// OC.linkTo matches the apps folders
+// OC.generateUrl ensure the index.php (or not)
+// We do not want the index.php since we're loading files
+// eslint-disable-next-line
+__webpack_public_path__ = generateFilePath('spreed', '', 'js/')
+
+Vue.prototype.t = translate
+Vue.prototype.n = translatePlural
+Vue.prototype.OC = OC
+Vue.prototype.OCA = OCA
+
+document.addEventListener('DOMContentLoaded', function() {
+
+	if (!window.OCA.Deck) {
+		return
+	}
+
+	window.OCA.Deck.registerCardAction({
+		label: t('spreed', 'Link to a conversation'),
+		icon: 'icon-talk',
+		callback: (card) => {
+			OCP.Collaboration.trigger('room').then(async(token) => {
+				const tempId = 'temp-' + card.id + '-' + token + '-' + (new Date().getTime())
+				try {
+					const response = await axios.post(generateOcsUrl('apps/spreed/api/v1', 2) + `chat/${token}/share`, {
+						objectType: 'deck-card',
+						objectId: card.id,
+						metaData: JSON.stringify(card),
+						referenceId: Hex.stringify(SHA1(tempId)),
+					})
+					const messageId = response.data.ocs.data.id
+					const targetUrl = generateUrl('/call/{token}#message_{messageId}', { token, messageId })
+					showSuccess(t('spreed', 'Deck card has been linked with the selected <a href="{link}">conversation</a>.', {
+						link: targetUrl,
+					}), {
+						isHTML: true,
+					})
+				} catch (exception) {
+					console.error('Error sharing deck card to conversation', exception, exception.response?.status)
+					showError(t('spreed', 'An error occurred while sharing deck card to conversation.'))
```

Comment:
Here you say sharing, maybe that's even a better word to use than link/post

Reply:
I hesitated a lot.
First I started with "sharing" because it was in the original mock up screenshot.
Then I saw "Link to conversation" in the hard-coded title of RoomSelector. At first believing that it was used elsewhere I simply went with that one. (note: we can't pass arguments to `trigger()` so have to settle for a window title globally)

Let's go with sharing then and I'll adjust the RoomSelector's title to also say so.
Or is that used elsewhere outside of Talk ?

Reply:
Meh... I'll go with "post", it sounds better.

Sharing makes it feel like you can unshare later.